boot.c

```diff
@@ -19,16 +20,10 @@ size_t utm_size;
 /* defined in entry.S */
 extern void* encl_trap_handler;
 
-#ifdef USE_FREEMEM
+/* defined in env.c */
+extern void* setup_start(void* _sp);
```

Comment:
Probably just create a header for setup_start and include that.

Reply:
Hmm, why do you prefer that?

Reply:
generally use extern for function prototypes only if you absolutely have to, it causes you to skip checks in the compiler that would otherwise catch various mistakes (ex: passing too few arguments etc)